connect/runtime/src/main/java/org/apache/kafka/connect/runtime/AbstractHerder.java

```diff
@@ -234,7 +231,7 @@ public ConfigInfos validateConnectorConfig(Map<String, String> connectorConfig)
             throw new BadRequestException("Connector config " + connectorConfig + " contains no connector type");
 
         Connector connector = getConnector(connType);
-
+        ClassLoader save = worker.getPlugins().compareAndSwapLoaders(connector);
```

Comment:
If I wanted to pick a single name that didn't make me happy, that was it (`save`). 
Not sure that a stack will make this part simpler. `pop` is implicit and in places I need to set the thread's classloader explicitly. Will check.